src/Manopt.jl

```diff
@@ -465,6 +465,11 @@ export SteepestDescentCoefficient,
     DaiYuanCoefficient,
     HagerZhangCoefficient,
     ConjugateGradientBealeRestart
+# Direction Corrections
+export AbstractRestartCondition
+export RestartNever,
+    RestartDescent,
+    RestartSufficientDescent
```

Comment:
I think naming wise I would prefer the grammatically easier to parse `NeverRestart` or even `NeverRestartCondition` for the first?